packages/codemirror/src/extension.ts

```diff
@@ -784,6 +792,55 @@ export namespace EditorExtensionRegistry {
           }
         }
       }),
+      Object.freeze({
+        name: 'extendSelection',
+        default: true,
+        factory: () =>
+          createConditionalExtension(
+            keymap.of([
+              {
+                key: 'Mod-Shift-l',
+                run: selectSelectionMatches,
+                preventDefault: true
+              },
+              { key: 'Mod-d', run: selectNextOccurrence, preventDefault: true }
```

Comment:
I tried quickly the (b) approach. But it is very complex to get the CM editor instance in the command in a generic way.

(a) sounds therefore the best path forward. But I did not go that road here because it will require more work than what I can afford.